src/pybind/ceph_argparse.py

```diff
@@ -1332,6 +1333,13 @@ def run_in_thread(func, *args, **kwargs):
     # strictly guaranteed is that *some* thread that has the signal
     # unblocked will receive it).  But there doesn't seem to be
     # any interface to create a thread with SIGINT blocked.
+
+    # Apparently the thread my not have fully exited after join returns,
+    # so implement a try-loop with backoff before claiming to timeout.
+    for i in range(0,100):
```

Comment:
after reading the source code of Python 3.6.10, i think this change practically prolong the "timeout" from `timeout` to `timeout + (0 + 99) * 100 / 100  / 2` => `timeout + 45.5` seconds.

could you share the python version you are using and output of of `ceph -w` with following patch applied?
```diff
diff --git a/src/pybind/ceph_argparse.py b/src/pybind/ceph_argparse.py
index 107f4d88fa..61f48adc61 100644
--- a/src/pybind/ceph_argparse.py
+++ b/src/pybind/ceph_argparse.py
@@ -1332,6 +1332,8 @@ def run_in_thread(func, *args, **kwargs):
     # strictly guaranteed is that *some* thread that has the signal
     # unblocked will receive it).  But there doesn't seem to be
     # any interface to create a thread with SIGINT blocked.
+    print('t._is_stopped = ', t._is_stopped)
+    print('t._started.set_set = ', t._started.is_set())
     if t.is_alive():
         raise Exception("timed out")
     elif t.exception:
```